### DISCUSSION TITLE
SK6812 fresh install 2023

### DISCUSSION THREAD

**Discussion by LaytexLlama** [awawa-dev/HyperHDR#493] [General]
Hi all,

I have a raspberry Pi 3B+, 10a 5v pwr supply, hdmi reader and a 5m roll of SK6812 leds.

I followed a YouTube video from V16 and where they used ws2812b leds.

Needless to say, my install did not work. I have read around but the more I read around the harder it seems to become. I can't find any clean explanation on how to install hyperhdr with SK6812 leds.

I was wondering if anyone would have the patience to help me out with what I need to do and where I went wrong. 1 better, link me to an existing explanation on start to finish, the install procedures for this situation.

I would like to add also for anyone willing to help that I'm actually an electrician (still learning these technical things though) so no need to fear about me zapping myself, I do know safety above all else :).

I also have a few esp32-wroom if they would help me out in this situation.

Many thanks in advance.

**Comment by awawa-dev** [comment]
> I followed a YouTube video from V16 and where they used ws2812b leds.

well, that guide is al least outdated.
get v19 beta2 SD image from releases section and install it. Backup your HyperHDR if you need it but it doesnt work anyway?

> I also have a few esp32-wroom if they would help me out in this situation.

Yes, they definitely should help. Probably all/most of them have an UART (like cp2104 etc) on the board and identifying these chips is also an important thing.

> I have a raspberry Pi 3B+, 10a 5v pwr supply, hdmi reader and a 5m roll of SK6812 leds.

Shared power supply? It's not a good think and may cause stability issues for the Rpi (https://github.com/awawa-dev/HyperHDR/discussions/320). 10a power supply is so so for 5m SK6812. It's below recommended max. specification although it may be sufficient (depends how the white component of LED is driven by software LED driver, there can be 4 RGBW components of the LED lit at max at the same but most algorithms use 'white' component subtraction). Measure the amperage when you set the 'white' color at full insensitivity for the safety.

Did test your LED setup and it works fine with WLED?

**Reply by LaytexLlama** [reply]
> well, that guide is al least outdated. Get v19 beta2 SD image from releases section and install it. Backup your HyperHDR if you need it but it doesnt work anyway?

Yeah, no point in me backing up the old version. I will just do a clean install with the newest and see how I go.


> Yes, they definitely should help. Probably all/most of them have an UART (like cp2104 etc) on the board and identifying these chips is also an important thing.

My ones have a CP2102 chip so all good there. How does the wiring work for using the ESP32 with the RPi? My exact model is ESP-WROOM-32. (pi is RPi 3B+)


> Shared power supply? It's not a good think and may cause stability issues for the Rpi (https://github.com/awawa-dev/HyperHDR/discussions/320)

I have now got a 5.1v 2.5a to dedicate to the RPi.


>  10a power supply is so so for 5m SK6812. It's below recommended max. specification although it may be sufficient

I dont plan on running the LEDs at 100% anyway. If there are issues with power I can just tone down the brightness.


> Did test your LED setup and it works fine with WLED?

I tested the 1.1m left on my 5m strip with WLED and they worked absolutely fine. White LEDs and RGB LEDs all worked.


Im just a little up in the air with where to begin, is there a specific installation guide for SK6812 or can I just follow the standard one? How does the ESP32 interact with the rest of the setup? Is there any other software other than HyperHDR I will need to make it all come together?


Thanks alot for you swift response last night and I look forward to bringing this all together and then trying to help others get it done too :)

**Comment by awawa-dev** [comment]
> My ones have a CP2102 chip so all good there. How does the wiring work for using the ESP32 with the RPi? My exact model is ESP-WROOM-32. (pi is RPi 3B+)

cp2102 is not optimal, I gave it only as an example but should be also fine. This chip is not capable of 2Mb speed when using serial connection with the USB/Adalight driver in HyperHDR and you must remember about it. Still you can adjust the speed and compile HyperSerialESP32 online (https://github.com/awawa-dev/HyperSerialESP32/wiki speed you want to use it here and in HyperHDR is 921600).  But I think you should first install version HyperSerialWLED 3.0 of https://github.com/awawa-dev/HyperSerialWLED  Choose v3 special version (921600) for ESP32 in the release: https://github.com/awawa-dev/HyperSerialWLED/releases/download/v3.0.0.0/special_version_speed_921600_only_HyperSerialWLED_ESP32.zip

In [HyperHDR configuration](https://user-images.githubusercontent.com/69086569/207109594-0493fe58-3530-46bb-a0a3-31a110475ed6.png) Use 921600 speed and please do not use 'white channel calibration (RGBW only)' because WLED doesn't supported it. It allows you to test the LED strip via wifi and using the serial connection from Rpi with the Serial/Adalight driver in HyperHDR.

Next if everything work fine, you should install HyperSerialESP32 with customized speed.

**Comment by awawa-dev** [comment]
Also you are missing important part of the setup: the level shifter https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter
Your LED strip is already "working" outside the technical specifiication but it may stop from time to time causing e.g. blinking (it mainly depends on how +5V from LED strip is stable, if it raises a bit over 5V under the load it shouldn't cause physical damage to the LED because there is still a limit but the 3.3.v data channel from Rpi/esp will become unrecognized for the LEDs).

**Comment by LaytexLlama** [comment]
So from the first comment, my understanding is that the ESP-32 should have (HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP32.bin) flashed on it?


> Also you are missing important part of the setup: the level shifter

I will be buying one of these tomorrow mornin :). 


Does the Pi and ESP32 talk together wirelessly?
Do I power the ESP32 off of the Pi?
![Quick HyperHDR diagram](https://user-images.githubusercontent.com/63335126/213912811-14e8694d-3643-4e0f-8a35-0ea6cf052455.png)


Could you please have a look at the image I've attached. Its just a dirty quick diagram (im sorry for how messy it is) of what goes where and all the rest. If you have something like this for a SK6812 specific build that would be awesome to work off.

**Reply by LaytexLlama** [reply]
I went and bought Fritzing also, I would love to make a proper layout once I have it all nutted out myself to make it easy in future for others, I just need some help getting there.

Thanks again for all your help and the awesome work you've done. Massively appreciated :D

**Comment by awawa-dev** [comment]
> So from the first comment, my understanding is that the ESP-32 should have (HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP32.bin) flashed on it?

Yes, to test if it works over wifi and over the serial connection. Then you can choose which method you prefer although the serial connection is better.

> Could you please have a look at the image I've attached. Its just a dirty quick diagram (im sorry for how messy it is) of what goes where and all the rest. If you have something like this for a SK6812 specific build that would be awesome to work off.

You need the common ground between ESP32&levelshifter and the LED strip only. You can power up ESP32 from Rpi when you connect it to the Rpi's USB port. For the level shifter you can use +5V from the LED strip power supply or from Rpi pinout (if ESP is connected to RPi USB) or from ESP32 (I think it's the Vin pin, but measure the resistance between the 5V pin and its power supply 5V, it should be 0 ohm), it doesn't matter.

**Reply by awawa-dev** [reply]
If you are not sure your LED power supply will provide stable +5V (fluctuating over 0.5V), then it's safer to use +5V for the level shifter from the LED strip power supply because the sk6812 logic voltage is related to VDD (LED strip +5v line) and allows +/- 0.5V error.
![obraz](https://user-images.githubusercontent.com/69086569/213925256-16149d16-4111-4d77-acbb-c46d8b7b96d7.png)

**Reply by LaytexLlama** [reply]
Just thought I'd ask considering I'll be waiting a while for the level shifter to arrive in Australia, what would be the ideal ESP board to get? Could you link me an optimal one to buy for this job?

**Reply by awawa-dev** [reply]
Here are examples of 2 recommended boards https://github.com/awawa-dev/HyperSerialESP32#example-of-supported-boards ESP32 MH-ET Live  is here for along time and it's proven solution, esp32-s2 is fresh solution.

**Reply by LaytexLlama** [reply]
Hello again,

Do you happen to have a guide for installing this? I cant seem to get my leds turning on (other than the first like 10)...
Below is what I'm using. If you could please link below the items what needs to be installed on each that would be great. Then maybe some screen grabs of the settings in HyperHDR.

Raspberry Pi 3B+ (With individual power supply)

The above mentioned ESP32 boards (powered from Rpi USB)

Capture Card (https://www.amazon.com.au/dp/B08GZ8CXQS?psc=1&ref=ppx_yo2ov_dt_b_product_details)

74AHCT125N Chip (If needed)

BTF-SK6812 60/m (With individual power supply)

Google TV 4K (going into the capture card, with individual power supply)

OLED 55 C9 TV

Im sorry for asking so many questions, all the guides I can find are so old and not sure if I should use them or what?

**Reply by awawa-dev** [reply]
>I cant seem to get my leds turning on (other than the first like 10)

did you test it using WLED interface, configured proper LED type/number and disabled power limit?

**Comment by LaytexLlama** [comment]
So I should first try get the LED's working with just WLED installed on the esp32?

**Reply by awawa-dev** [reply]
I dont know, was only guesing...you didnt describe how did you test your setup.

**Comment by LaytexLlama** [comment]
I installed (HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP32.bin) on the esp32. Also installed the most recent hyperhdr to the Rpi. Other than that it's wired exactly as you said and in the diagram above

**Comment by awawa-dev** [comment]
You need to configure wled first (with phone over wifi, there are lots of tutorials how to connect to it) even if you plan to use it with the serial connection only.
Pay attention to these things:
> did you test it using WLED interface, configured proper LED type/number and disabled power limit?

**Comment by awawa-dev** [comment]
And `HyperSerialWLED_special_version_speed_921600_nodemcuv2_ESP32.bin` is almost regular WLED 0.12 edition with improved serial communication. Must be configured and tested with your LED strip as with the normal WLED version first before proceeding to the next step (using a serial communication over USB).
